packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx

```diff
@@ -126,13 +126,12 @@ const VisitHeader: React.FC = () => {
 
   const noActiveVisit = !isLoading && visitNotLoaded;
 
-  const originPage = localStorage.getItem('fromPage');
-
   const onClosePatientChart = useCallback(() => {
-    originPage ? navigate({ to: `${window.spaBase}/${originPage}` }) : navigate({ to: `${window.spaBase}/home` });
+    const originPage = window.localStorage.getItem('patient-chart-origin-page');
```

Comment:
This may not be the point of this PR, but caching this data in local storage is not the right thing to do. Session storage would be slightly more appropriate, but really, I'd just store it in memory (i.e., as a variable or piece of state.

Reply:
Thanks for pointing the scenario if the user opens multiple tabs.
I'll try to test it then.
Thanks!